test/templates/template_api_test.rb

```diff
@@ -18,16 +18,22 @@ def setup
 
   def test_api_can_return_templateserver
     get "/templateServer"
-    assert last_response.ok?, "Last response was not ok"
+    assert last_response.ok?, "Last response was ok"
     data = JSON.parse(last_response.body)
     assert_equal("someproxy:8443", data["templateServer"].to_s)
   end
 
   def test_api_can_ask_for_a_template
     Proxy::Templates::TemplateProxyRequest.any_instance.expects(:get_template).returns("An API template")
     get "/provision", @args
-    assert last_response.ok?, "Last response was not ok"
+    assert last_response.ok?, "Last response was ok"
     assert_match("An API template", last_response.body)
   end
 
+  def test_api_can_ask_for_a_hostgroup_template
+    Proxy::Templates::TemplateProxyRequest.any_instance.expects(:get_template).with("kind/temp/hg", anything, anything).returns("An API template")
+    get "/kind/temp/hg", @args
```

Comment:
I don't see any verification (here and in the test above) that the contents of @args is being passed into TemplateProxyRequest. Ideally, there should be a dedicated test that verifies that request parameters are being passed further along. At the very least, could you skip args or pass an empty hash in these two tests instead?